### PR TITLE
ADB Cheat Sheet

### PR DESCRIPTION
**Pull request by gauthamyu5010**
DDG Community Kochi
#2127 
IA Link:https://duck.co/ia/view/adb_cheatsheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@gauthamyu5010 Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

While the team and the community review your submission, we'd like to make sure you look good! If you'd like a free DuckDuckGo tshirt, please send over the following info to merch@duckduckgo.com, and be sure to put "Dev Merch" in the subject line:

Name:
Street Address:
City:
State / Province:
Country:
Phone (Non-US Orders only): 
T-shirt size:
GitHub username:
Link to this Pull Request:

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=ADB Cheatsheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/adb_cheatsheet_mobile.png)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=ADB Cheatsheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/adb_cheatsheet_mobile.png)

File: share/goodie/cheat_sheets/json/adb.json [hunk 1]
```diff
@@ -0,0 +1,68 @@
+{
+  "id": "ADB_cheat_sheet",
+  "name": "ADB",
+  "description": "Android SDK Tool",
+  "metadata": {
+    "sourceName": "ADB",
+    "sourceURL": "https://github.com/maldroid/adb_cheatsheet"
+  },
+  "aliases": [
+    "ADB_cheat_sheet"
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/adb.json]: @gauthamyu5010 Avoid upper case letters in aliaes.
No need to use under score. Use spaces where needed.
"cheat sheet" is already a triggering word for cheat sheets .So avoid using such words(cheat sheet,commands,help etc.) in aliases

File: share/goodie/cheat_sheets/json/adb.json [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  "template_type": "keyboard",
+  "section_order": [
+    "devices by serial",
+    "devices by model",
+    "push",
+    "backup",
+    "log",
+    "kill-server",
+    "start-server",
+    "root",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/adb.json]: @gauthamyu5010 lets capitalise first letter of each word in section names.
Also remove the hipens

File: share/goodie/cheat_sheets/json/adb.json [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      {
+        "key": "[Ctrl] [Alt] [P]",
+        "val": "Copy le/dir to device."
+      }
+    ],
+    "log": [
+      {
+        "key": "[Alt] [L]",
+        "val": "pull logcat"
+      }
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/adb.json]: @gauthamyu5010 lets start the val field with a capital letter.
some of above val field also have to be corrected from the same

File: share/goodie/cheat_sheets/json/adb.json [hunk 4]
```diff
@@ -0,0 +1,68 @@
+{
+  "id": "ADB_cheat_sheet",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/adb.json]: please lowercase this: `adb_cheat_sheet`

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=ADB Cheatsheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/adb_cheatsheet_mobile.png)

**Comment by edgesince84** [comment]
@gauthamyu5010,  thanks again for (creating/fixing) the ADB Cheat Sheet! You’ve been doing a really good job here and we'd like you to help our community tackle the next big challenge - our mission to become the best search engine for programmers: https://duck.co/blog/post/297/help-for-programmers.

When you're ready to move on, please: 

- Create a new Instant Answer to increase our Programming IA coverage

OR

- Test live programming Instant Answers (or those in development) for bugs and suggest improvements 

Detailed instructions explaining how you can help are available at the top of this [page](https://github.com/duckduckgo/duckduckgo/wiki/Programming-IA-Coverage).

Thanks again!

**Comment by GuiltyDolphin** [comment]
@gauthamyu5010 I'm closing this for now - if you want to continue just give us a ping and we'll re-open it!

**Pull request closed without merging**